components/Header.tsx

```diff
@@ -1,74 +1,130 @@
+    },
+    { name: "Submit a project", href: "/submit", class: "sm:hidden" },
+  ];
+
   return (
-    <header class={SITE_BAR_STYLES}>
-      <a href="/">
-        <img
-          height="48"
-          width="48"
-          src="/logo.webp"
-          alt={SITE_NAME + " logo"}
-          class="h-12 w-12"
-        />
-      </a>
-      <nav class={NAV_STYLES}>
-        {stripe
-          ? (
-            <a
-              href="/pricing"
-              class={getToggledStyles(
-                LINK_STYLES,
-                ACTIVE_LINK_STYLES,
-                props.url.pathname === "/pricing",
-              )}
-            >
-              Pricing
-            </a>
-          )
-          : null}
-        {props.sessionId
-          ? (
-            <a
-              href="/account"
-              class={getToggledStyles(
-                LINK_STYLES,
-                ACTIVE_LINK_STYLES,
-                props.url.pathname === "/account",
-              )}
-            >
-              Account
-            </a>
-          )
-          : <a href="/signin" class={LINK_STYLES}>Sign in</a>}
-        <a
-          href="/account/notifications"
-          class={getToggledStyles(
-            LINK_STYLES,
-            ACTIVE_LINK_STYLES,
-            props.url.pathname === "/account/notifications",
-          ) + " relative"}
-          aria-label="Notifications"
-        >
-          <Bell class="w-6 h-6" />
-          {props.hasNotifications && (
-            <CircleFilled class="absolute top-0.5 right-0.5 text-primary w-2 h-2" />
-          )}
+    <header
+      class={cx(
+        SITE_BAR_STYLES,
+        "flex-col sm:flex-row",
+      )}
+    >
+      <input
+        type="checkbox"
+        id="nav-toggle"
+        class="hidden [:checked&+*>:last-child>*>:first-child]:hidden [:checked&+*>:last-child>*>:last-child]:block checked:siblings:last-child:flex"
+      />
+
+      <div class="flex justify-between items-center">
+        <a href="/" class="shrink-0">
+          <img
+            height="48"
+            width="48"
+            src="/logo.webp"
+            alt={SITE_NAME + " logo"}
+            class="h-12 w-12"
+          />
         </a>
-        <div class="rounded-lg bg-gradient-to-tr from-secondary to-primary p-px">
+        <div class="flex gap-4 items-center">
+          <label
+            tabIndex={0}
+            class="sm:hidden"
+            id="nav-toggle-label"
+            htmlFor="nav-toggle"
+          >
+            <Bars class="w-6 h-6" />
+            <Cross class="hidden w-6 h-6" />
+          </label>
+        </div>
+      </div>
+      <script>
+        {`
+          const navToggleLabel = document.getElementById('nav-toggle-label');
+          navToggleLabel.addEventListener('keydown', () => {
+            if (event.code === 'Space' || event.code === 'Enter') {
+              navToggleLabel.click();
+              event.preventDefault();
+            }
+          });
+        `}
+      </script>
```

Comment:
Would it be better to have the menu as an island? Genuine question.

Reply:
I'm a little unsure of this. In [the code I referenced](https://github.com/denoland/doc_components/blob/c46d3cdc9505917f5681805f13111f311779a0c6/header.tsx#L101), we wrote it directly in the onClick handler of the element, but since this caused an error in the latest version of Fresh, I wrote it like that.

If it is set to Island, it will take some time before it can be operated, and it is necessary to consider the time when hydration is not done.

In this case, I just wanted to improve accessibility and didn't plan to add anything here, it seems bit overkill to make it an island, so I put it inline. but I don't have a strong opinion here.

Reply:
Fair enough. I didn't feel strongly either.